gips/test/unit/t_cdl.py

```diff
@@ -0,0 +1,33 @@
+import datetime
+import os
+
+import pytest
+
+from ...data.cdl import cdl
+
+_2016 = datetime.date(2016, 1, 1)
+_afps = {
+    'cdl': '/repo/cdl/tiles/IN/2016/IN_2016_cdl_cdl.tif',
+    'cdlmkii': '/repo/cdl/tiles/IN/2016/IN_2016_cdlmkii_cdl.zip',
+}
+
+def _patched_cdlData_init(self, *args, **kwargs):
+    asset = cdl.cdlAsset(os.path.dirname(__file__) + '/../IN_2016_cdlmkii_cdl.zip')
+    self.assets = {'cdlmkii': asset}
+
+asset_constructor_base_case_params = [
+    (asset_fp, (asset_fp, ap_type, 'IN', _2016, 'cdl', {ap_type: asset_fp}))
+    for ap_type, asset_fp in _afps.items()
+]
+
+@pytest.mark.parametrize("fn, expected", asset_constructor_base_case_params)
+def t_cdlAsset_constructor(fn, expected):
+    a = cdl.cdlAsset(fn)
+    assert expected == (a.filename, a.asset, a.tile, a.date, a.sensor, a.products)
+
+def t_cdlData_legend(mocker):
+    mocker.patch('gips.data.cdl.cdlData.__init__', _patched_cdlData_init)
+    d = cdl.cdlData()
+    assert d.legend() == ['class1', 'class2']
```

Comment:
optional:  one of the rules of automated testing is supposed to be only doing one assertion per test.  This is because the first assert failure masks the rest.  So you might as well get all the info you can about what happened, it may help.  So build them into a single assert (`assert expr and expr and expr`).  Note I cheat all the time at this in unit tests as it's tough to mix `some_mock.assert_called_with(arg, arg)` and `assert some_expr`.